db/db_impl/db_impl_open.cc

```diff
@@ -129,6 +129,10 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
   for (size_t i = 0; i < result.db_paths.size(); i++) {
     DeleteScheduler::CleanupDirectory(result.env, sfm, result.db_paths[i].path);
   }
+  if (!result.wal_dir.empty()) {
```

Comment:
Actually, ```wal_dir``` will not be empty as it is defaulted to dbname in line 95 if the user didn't set it. So maybe just compare it to dbname.